registrations/views.py

```diff
@@ -625,3 +632,140 @@ def increment_position(self, request, pk=None):
+            identity = self.identity_store.get_identity_by_address("msisdn", msisdn)
+            return next(identity["results"])
+        # Catch both no results and HTTP errors
+        except (StopIteration, RequestException):
+            return None
+
+    def get_registrations(self, identity):
+        """
+        Gets the list of registrations for the identity, ordered from oldest to newest.
+        If there is no identity, returns empty list
+        """
+        try:
+            identity_id = identity["id"]
+        except (KeyError, TypeError):
+            return []
+
+        return Registration.objects.filter(registrant_id=identity_id).order_by(
+            "created_at"
+        )
+
+    def lookup_field_from_dictionaries(self, field, *dictionaries):
+        """
+        Given a field and one or more dictionaries, returns the first match that it
+        finds, or None if no match is found
+        """
+        for dictionary in dictionaries:
+            if field in dictionary:
+                return dictionary[field]
+
+    def extract_info(self, identity, registrations):
+        """
+        Extracts the info from the identity and registrations to be displayed.
+        Returns a map of labels and values.
+        """
+        result = {}
+
+        registrations_details = []
+        for registration in registrations:
+            result["Registration Type"] = registration.get_reg_type_display()
+            registrations_details.append(registration.data)
+
+        # Reverse registration details so that we get latest information first
+        registrations_details = registrations_details[::-1]
+
+        try:
+            identity_details = identity["details"]
+        except (KeyError, TypeError):
+            identity_details = {}
+
+        for field, label in self.LOOKUP_FIELDS:
+            value = self.lookup_field_from_dictionaries(
+                field, identity_details, *registrations_details
+            )
+            if value:
+                result[label] = value
+
+        return result
+
+    def post(self, request):
+        self.validate_signature(request)
+
+        serializer = self.get_serializer_class()(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        context = []
+
+        msisdn = self.get_msisdn(serializer.validated_data)
+        identity = self.get_identity(msisdn)
+        registrations = self.get_registrations(identity)
+        info = self.extract_info(identity, registrations)
+        if info:
+            context.append(
+                {
+                    "icon": "info-circle",
+                    "title": "Mother's Details",
+                    "type": "table",
+                    "timestamp": timezone.now().isoformat(),
+                    "uuid": "7eb448be-3180-417e-823c-0c6d5da24e00",
+                    "payload": info,
+                }
+            )
+
+        return Response({"version": "1.0.0-alpha", "context": context})
```

Comment:
Not sure about this version? Shouldn't it be more dynamic.

Reply:
This is us sending it to engage, according to the API docs.

I'm not entirely sure what it is for, but my guess is that if engage ever changes this API, then it will identify and understand the old format by this string, and then when we upgrade to the new format we change this string to show that we're now sending in the new format, basically kind of like a "message format". So we want it to be static, if I'm understanding correctly.